modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java

```diff
@@ -210,6 +210,72 @@ public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO) throws
 		return appDtoResult;
 	}
 
+	/**
+	 * Create Application with a given appDTO and service provider name
+	 *
+	 * @return OAuthConsumerAppDTO
+	 * @throws Exception
+	 */
+	public OAuthConsumerAppDTO createApplication(OAuthConsumerAppDTO appDTO, String serviceProviderName) throws Exception {
+		OAuthConsumerAppDTO appDtoResult = null;
+
+		adminClient.registerOAuthApplicationData(appDTO);
+		OAuthConsumerAppDTO[] appDtos = adminClient.getAllOAuthApplicationData();
+
+		for (OAuthConsumerAppDTO appDto : appDtos) {
+			appDtoResult = appDto;
+			consumerKey = appDto.getOauthConsumerKey();
+			consumerSecret = appDto.getOauthConsumerSecret();
+		}
+		ServiceProvider serviceProvider = new ServiceProvider();
+		serviceProvider.setApplicationName(serviceProviderName);
+		serviceProvider.setDescription(SERVICE_PROVIDER_DESC);
+		serviceProvider.setManagementApp(true);
+		appMgtclient.createApplication(serviceProvider);
+
+		serviceProvider = appMgtclient.getApplication(serviceProviderName);
+		serviceProvider = setServiceProviderClaimConfig(serviceProvider);
+		serviceProvider.setOutboundProvisioningConfig(new OutboundProvisioningConfig());
+		List<InboundAuthenticationRequestConfig> authRequestList = new ArrayList<>();
+
+		if (consumerKey != null) {
+			InboundAuthenticationRequestConfig opicAuthenticationRequest = new InboundAuthenticationRequestConfig();
+			opicAuthenticationRequest.setInboundAuthKey(consumerKey);
+			opicAuthenticationRequest.setInboundAuthType("oauth2");
```

Comment:
I am guess you meant OIDC here?

Reply:
Fixed in https://github.com/wso2/product-is/pull/14727/commits/05582505dc2e6544c8721e447305e82b003e86f6